### PR TITLE
Clean up base encoder class code.

### PR DESCRIPTION
**Pull request by scottpurdy**
Fixes: #1945

@chetan51 please review

### CODE REVIEW

**Comment by chetan51** [comment]
Guess you couldn't cheat the Fixes Issue Validator, by trying to link this PR to itself :)

File: nupic/encoders/base.py [hunk 1]
```diff
@@ -39,78 +38,74 @@
...
-  ############################################################################
   def setLearning(self, learningEnabled):
-    """
-    Set whether learning is enabled.
+    """Set whether learning is enabled.
 
     @param learningEnabled whether learning should be enabled
     """
-    if hasattr(self,"_learningEnabled"):
+    # TODO: Make sure subclasses don't rely on this and remove it. Default
```
Comment (chetan51) [inline nupic/encoders/base.py]: This should be referenced in an open issue.

**Comment by chetan51** [comment]
Other than the one comment, :+1:

**Comment by rhyolight** [comment]
:rage:

**Pull request merged**